File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
from collections import deque
def solution(people, limit):
    count = 0
    li = deque(sorted(people))
    while len(li) > 0:
        if len(li) == 1:
            count += 1
            break
            
        if li[0] + li[-1] <= limit:
            li.popleft()
            li.pop()
            
        else:
            li.pop()
            
        count += 1
    return count

File: 프로그래머스/lv2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        Deque<Integer> li = new ArrayDeque<>();
        for(int i : people){
            li.add(i);
        }
        
        int answer = 0;
        while(!li.isEmpty()){
            if(li.size() == 1){
                answer++;
                break;
            }
            if(li.getFirst() + li.getLast() <= limit){
                li.removeFirst();
                li.removeLast();
            } else {
                li.removeLast();
            }
            answer++;
        }
        return answer;
    }
}